pkg/init/backend/interactive.go

```diff
@@ -104,3 +106,149 @@ func (o *InteractiveBackend) PersonalizeName(devfile parser.DevfileObj, flags ma
+		PrintConfiguration(config)
+		selectContainerAnswer, err = o.askerClient.AskContainerName(containerOptions)
+		if err != nil {
+			return err
+		}
+
+		selectedContainer := config[selectContainerAnswer]
+		if selectContainerAnswer == "NONE - configuration is correct" {
+			break
+		}
+
+		var configOps asker.OperationOnContainer
+		for configOps.Ops != "Nothing" {
+			configOps, err = o.askerClient.AskPersonalizeConfiguration(selectedContainer)
+			if err != nil {
+				return err
+			}
+			switch configOps.Ops {
+			case "Add":
+				switch configOps.Kind {
+				case "Port":
+					var newPort string
+					newPort, err = o.askerClient.AskAddPort()
+					if err != nil {
+						return err
+					}
+
+					err = devfileobj.Data.SetPorts(map[string][]string{selectContainerAnswer: {newPort}})
+					if err != nil {
+						return err
+					}
+					selectedContainer.Ports = append(selectedContainer.Ports, newPort)
+
+				case "EnvVar":
+					var newEnvNameAnswer, newEnvValueAnswer string
+					newEnvNameAnswer, newEnvValueAnswer, err = o.askerClient.AskAddEnvVar()
+					if err != nil {
+						return err
+					}
+					err = devfileobj.Data.AddEnvVars(map[string][]v1alpha2.EnvVar{selectContainerAnswer: {{
+						Name:  newEnvNameAnswer,
+						Value: newEnvValueAnswer,
+					}}})
+					if err != nil {
+						return err
+					}
+					selectedContainer.Envs[newEnvNameAnswer] = newEnvValueAnswer
+				}
+			case "Delete":
+				switch configOps.Kind {
+				case "Port":
+					portToDelete := configOps.Key
+					indexToDelete := -1
+					for i, port := range selectedContainer.Ports {
+						if port == portToDelete {
+							indexToDelete = i
+						}
+					}
+					if indexToDelete == -1 {
+						log.Warningf(fmt.Sprintf("unable to delete port %q, not found", portToDelete))
+					}
+					err = devfileobj.Data.RemovePorts(map[string][]string{selectContainerAnswer: {portToDelete}})
+					if err != nil {
+						return err
+					}
+					selectedContainer.Ports = append(selectedContainer.Ports[:indexToDelete], selectedContainer.Ports[indexToDelete+1:]...)
+
+				case "EnvVar":
+					envToDelete := configOps.Key
+					if _, ok := selectedContainer.Envs[envToDelete]; !ok {
+						log.Warningf(fmt.Sprintf("unable to delete env %q, not found", envToDelete))
+					}
+					err = devfileobj.Data.RemoveEnvVars(map[string][]string{selectContainerAnswer: {envToDelete}})
+					if err != nil {
+						return err
+					}
+					delete(selectedContainer.Envs, envToDelete)
+				}
+			case "Nothing":
+			default:
+				return fmt.Errorf("Unknown configuration selected %q", fmt.Sprintf("%v %v %v", configOps.Ops, configOps.Kind, configOps.Key))
+			}
+			// Update the current configuration
+			config[selectContainerAnswer] = selectedContainer
+		}
+	}
+	return devfileobj.WriteYamlDevfile()
+}
+
+func PrintConfiguration(config asker.DevfileConfiguration) {
+	color.New(color.Bold, color.FgGreen).Println("Current component configuration:")
+
+	for key, container := range config {
```

Comment:
`config` being a map, the elements will be rendered in a random order every time. I think it could be more readable by sorting the keys first